lib/nexmo_markdown_renderer/filters/snippet_variables_filter.rb

```diff
@@ -0,0 +1,33 @@
+module Nexmo
+  module Markdown
+    class SnippetVariablesFilter < Banzai::Filter
+      def call(input)
+        input.gsub(/```snippet_variables(.+?)```/m) do |_s|
+          config = YAML.safe_load(Regexp.last_match(1))
+
+          raise 'No variables provided' unless config
+          raise 'Must provide a list' unless config.is_a?(Array)
+
+          output = <<~HEREDOC
+            Key | Description
+            -- | --
+          HEREDOC
+          config.each do |key|
+            details = variables[key]
+            raise "#{key} is not a valid snippet variable" unless details
```

Comment:
Should we add another exception if the `key` does exist, but the `description` parameter is not present?

Reply:
Test added, good idea!